lib/utils.js

```diff
@@ -3,49 +3,33 @@ const nqlLang = require('@nexes/nql-lang');
 
 /**
  * ## Merge Filters
- * Util to combine overrides, default and custom filters based on
- * following hierarchy: overrides -> custom -> defaults
- *
- * overrides - filters which must ALWAYS be applied
- * defaults - filters which must be applied if a matching filter isn't provided
- * custom - custom filters which are additional
+ * Util to combine multiple filters based on the priority how
+ * they are passed into the method. For example:
+ *      mergeFilter(overrides, custom, defaults);
+ * would merge these three filters having overrides on highers priority
+ * and defaults on the lowest priority
  *
  * @TODO:
  *
- * - make function signature more generic to accept any amount of arguments/filters
- * - and then move the function to mongo-utils?
+ * - the function to mongo-utils?
  */
-const mergeFilters = ({overrides, defaults, custom} = {}) => {
-    if (custom && !overrides && !defaults) {
-        return custom;
-    }
-
+const mergeFilters = (...filters) => {
```

Comment:
@kirrg001 I tried this change with one case just to see how hard it actually is. It was so easy I just went for the change. Let me know if this makes sense :) All I had to do is reorder the arguments so they are passed within their priority, the core logic didn't have to change at all, just remove unnecessary if's

Reply:
Also, this method became so generic and small don't think it should stay here anymore and could easily go into mongo-utils?

Reply:
100%
Did you double check if all Ghost tests still pass with this refactoring?

Reply:
Maybe move the merge fn before merging, it's easy peasy.

Reply:
Cool :sunglasses: Will do it quickly in the morning tomorrow. How do you think the call to `mergeFilters` should look like from `nql.js`, should it:
- go througt it's own utils (wrapping function that just exports `('mongo-utils').mergeFilters`
or
- should it import mongo-utils in `nql.js` file directly? 

Kind of leaning to the former option personally to keep less imports in `nql.js`

Reply:
> Did you double check if all Ghost tests still pass with this refactoring?
Haven't checked but pretty sure it works as the tests cover pretty much all of the usecases. But will do before merging in

Reply:
> Kind of leaning to the former option personally to keep less imports in nql.js

👍

Reply:
Have moved mergeFilters to mongo-utils. Tested these changes in Ghost and all functional tests pass :100: